actions/online-sign/action.yml

```diff
@@ -0,0 +1,71 @@
+name: 'Online sign'
+description: 'Creates a snapshot and timestamp if needed, moves publish branch if needed'
+
+inputs:
+  force_publish:
+    description: 'Publish even when snapshot and timestamp are not created'
+    required: true
+  gcp_workload_identity_provider:
+    description: 'Google Cloud workload identity provider'
+    required: false
+    default: ''
+  gcp_service_account:
+    description: 'Google Cloud service account name'
+    required: false
+    default: ''
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+      with:
+        fetch-depth: 0
+
+    - name: Authenticate to Google Cloud
+      if: inputs.gcp_workload_identity_provider != ''
+      uses: google-github-actions/auth@35b0e87d162680511bf346c299f71c9c5c379033 # v1.1.1
+      with:
+        token_format: access_token
+        workload_identity_provider: ${{ inputs.gcp_workload_identity_provider }}
+        service_account: ${{ inputs.gcp_service_account }}
+
+    - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4.7.0
+      with:
+        python-version: "3.11"
+
+    - run: pip install $GITHUB_ACTION_PATH/../../repo/
+      shell: bash
+
+    - id: online-sign
+      run: |
+        tuf-on-ci-online-sign --push
+
+        # did we actually create a snapshot/timestamp commit?
+        if [[ $GITHUB_SHA == $(git rev-parse HEAD) ]]; then
+          echo "ONLINE_SIGNED=false"
+          echo "ONLINE_SIGNED=false" >> "$GITHUB_ENV"
+        else
+          echo "ONLINE_SIGNED=true"
+          echo "ONLINE_SIGNED=true" >> "$GITHUB_ENV"
+        fi
+      shell: bash
+
+    - id: move-publish-branch
+      if: inputs.force_publish == 'true' || env.ONLINE_SIGNED == 'true'
+      run: |
+        git show --oneline --no-patch HEAD
+        git push origin HEAD:publish
+      shell: bash
```

Comment:
moving the publish branch currently happens here, in the step following the `tuf-on-ci-online-sign` call.

The alternative would be to do it in `tuf-on-ci-online-sign` (this would make some sense as it already pushes to "main"). The only thing needed for that is forwarding `force_publish` as an argument.

I think this alternative makes more sense as then the git pushes are in a single place

Reply:
Without thinking too much, I prefer to have the push to publish separated into a different steps, it feels like we are having better separation of duties this way? The command always work against `main`, but the workflows are in charge of cordinate all the steps?

Reply:
Yeah that is the thinking that lead me to write it this way: that pushing to "publish" is a separate decision from pushing to "main"...

But after implementing it I'm not so sure: 
* there's no checks I can think of that would make sense to do at this point (that wouldn't also make sense before merging to main) and there is a separate publish workflow already.
* keeping functionality in the workflow (instead of the tool) makes it harder to test

So I think pushing to "publish" at the same time as pushing to "main" would make more sense... but this is not a strong opinion.

Reply:
I will resolve this without changes:
* Changing this is not directly necessary for this PR: this is now an implementation detail of `online-sign` action and we can improve this later
* I think the change makes sense... but it's not actually trivial: I would rather have it separately reviewed if it's done